cms/djangoapps/contentstore/rest_api/v0/views/tabs.py

```diff
@@ -0,0 +1,239 @@
+                "settings": {},
+                "tab_id": "courseware",
+                "title": "Course",
+                "type": "courseware"
+            },
+            ...
+        }
+        ```
+        """
+        course_key = CourseKey.from_string(course_id)
+        if not has_studio_read_access(request.user, course_key):
+            raise self.permission_denied(request)
+
+        course_module = modulestore().get_course(course_key)
+        tabs_to_render = get_course_tabs(course_module, request.user)
+        return Response(CourseTabSerializer(tabs_to_render, many=True).data)
+
+
+@view_auth_classes(is_authenticated=True)
+class CourseTabSettingsView(DeveloperErrorViewMixin, APIView):
+    """
+    API view for course tabs settings.
+    """
+
+    def handle_exception(self, exc):
+        """Handle NotImplementedError and return a proper response for it."""
+        if isinstance(exc, NotImplementedError):
+            return self._make_error_response(400, str(exc))
+        if isinstance(exc, ItemNotFoundError):
+            return self._make_error_response(400, str(exc))
+        return super().handle_exception(exc)
+
+    @apidocs.schema(
+        body=CourseTabUpdateSerializer(help_text=_("Change the visibility of tabs in a course.")),
+        parameters=[
+            apidocs.string_parameter("course_id", apidocs.ParameterLocation.PATH, description="Course ID"),
+            apidocs.string_parameter("tab_id", apidocs.ParameterLocation.QUERY, description="Tab ID"),
+            apidocs.string_parameter("tab_location", apidocs.ParameterLocation.QUERY, description="Tab usage key"),
+        ],
+        responses={
+            204: "In case of success, a 204 is returned with no content.",
+            401: "The requester is not authenticated.",
+            403: "The requester cannot access the specified course.",
+            404: "The requested course does not exist.",
+        },
+    )
+    @verify_course_exists
+    def post(self, request: Request, course_id: str) -> Response:
+        """
+        Change visibility of tabs in a course.
+
+        **Example Requests**
+
+        You can provide either a tab_id or a tab_location.
+
+        Hide a course tab using ``tab_id``:
+
+            POST /api/contentstore/v0/tabs/{course_id}/settings/?tab_id={tab_id} {
+                "is_hidden": true
+            }
+
+        Hide a course tab using ``tab_location``
+
+            POST /api/contentstore/v0/tabs/{course_id}/settings/?tab_location={tab_location} {
+                "is_hidden": true
+            }
+
+        **Response Values**
+
+        If the request is successful, an HTTP 204 response is returned
+        without any content.
+        """
+        course_key = CourseKey.from_string(course_id)
+        if not has_studio_write_access(request.user, course_key):
+            raise self.permission_denied(request)
+
+        tab_id_locator = TabIDLocatorSerializer(data=request.query_params)
```

Comment:
what value are we expecting in `request.query_params`?

Reply:
I've documented that in the docstring for this function, and the swagger schema. It matches the serializer used here.

Reply:
Got it, thanks,